dev/CommandBarFlyout/CommandBarFlyoutCommandBar.cpp

```diff
@@ -475,6 +659,433 @@ void CommandBarFlyoutCommandBar::UpdateTemplateSettings()
     }
 }
 
+void CommandBarFlyoutCommandBar::EnsureAutomationSetCountAndPosition()
+{
+    COMMANDBARFLYOUT_TRACE_VERBOSE(*this, TRACE_MSG_METH, METH_NAME, this);
+
+    auto moreButton = m_moreButton.get();
+    int sizeOfSet = 0;
+
+    for (const auto& command : PrimaryCommands())
+    {
+        if (auto commandAsUIElement = command.try_as<winrt::UIElement>())
+        {
+            if (commandAsUIElement.Visibility() == winrt::Visibility::Visible)
+            {
+                sizeOfSet++;
+            }
+        }
+    }
+
+    if (moreButton)
+    {
+        // Accounting for the MoreButton
+        sizeOfSet++;
+    }
+
+    for (const auto& command : PrimaryCommands())
+    {
+        if (auto commandAsUIElement = command.try_as<winrt::UIElement>())
+        {
+            if (commandAsUIElement.Visibility() == winrt::Visibility::Visible)
+            {
+                winrt::AutomationProperties::SetSizeOfSet(commandAsUIElement, sizeOfSet);
+            }
+        }
+    }
+
+    if (moreButton)
+    {
+        winrt::AutomationProperties::SetSizeOfSet(moreButton, sizeOfSet);
+        winrt::AutomationProperties::SetPositionInSet(moreButton, sizeOfSet);
+    }
+}
+
+void CommandBarFlyoutCommandBar::EnsureFocusedPrimaryCommand()
+{
+    COMMANDBARFLYOUT_TRACE_VERBOSE(*this, TRACE_MSG_METH, METH_NAME, this);
+
+    MUX_ASSERT(!SharedHelpers::IsRS3OrHigher());
+
+    auto moreButton = m_moreButton.get();
+    auto tabStopControl = GetFirstTabStopControl(PrimaryCommands());
+
+    if (!tabStopControl)
+    {
+        if (moreButton && moreButton.IsTabStop())
+        {
+            tabStopControl = moreButton;
+        }
+    }
+
+    if (tabStopControl)
+    {
+        if (tabStopControl.FocusState() == winrt::FocusState::Unfocused)
+        {
+            FocusControl(
+                tabStopControl /*newFocus*/,
+                nullptr /*oldFocus*/,
+                winrt::FocusState::Programmatic /*focusState*/,
+                false /*updateTabStop*/);
+        }
+    }
+    else
+    {
+        FocusCommand(
+            PrimaryCommands() /*commands*/,
+            moreButton /*moreButton*/,
+            winrt::FocusState::Programmatic /*focusState*/,
+            true /*firstCommand*/,
+            true /*ensureTabStopUnicity*/);
+    }
+}
+
+void CommandBarFlyoutCommandBar::OnKeyDown(
```

Comment:
Is this actually used anywhere?  Ctrl+F on this page says no, but I might've missed it.

Reply:
Yep - it does get called by the framework.

Reply:
Ohh, is this an override?  Got it, I was expecting this to be passed in as en event handler somewhere.